code/aepcomposeui/src/main/java/com/adobe/marketing/mobile/aepcomposeui/interactions/UIEevent.kt

```diff
@@ -0,0 +1,49 @@
+package com.adobe.marketing.mobile.aepcomposeui.interactions
+
+import com.adobe.marketing.mobile.aepuitemplates.AEPUITemplate
+import com.adobe.marketing.mobile.aepcomposeui.aepui.AEPUI
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.AEPUIState
+
+/**
+ * Sealed interface representing different types of UI events in AEP.
+ *
+ * @param T The type of UI template associated with the event.
+ * @param S The type of UI state associated with the event.
+ */
+sealed interface UIEvent<T : AEPUITemplate, S : AEPUIState> {
+    /**
+     * Event that represents the display of a UI element.
+     *
+     * @param T The type of UI template associated with the display event.
+     * @param S The type of UI state associated with the display event.
+     * @property _aepui The AEPUI associated with the display event.
+     */
+    data class Display<T : AEPUITemplate, S : AEPUIState>(val _aepui: AEPUI<T, S>) : UIEvent<T, S>
+
+    /**
+     * Event that represents a click interaction with a UI element.
+     *
+     * @param T The type of UI template associated with the click event.
+     * @param S The type of UI state associated with the click event.
+     * @property _aepui The AEPUI associated with the click event.
+     */
+    data class Click<T : AEPUITemplate, S : AEPUIState>(val _aepui: AEPUI<T, S>) : UIEvent<T, S>
+
+    /**
+     * Event that represents the dismissal of a UI element.
+     *
+     * @param T The type of UI template associated with the dismiss event.
+     * @param S The type of UI state associated with the dismiss event.
+     * @property _aepui The AEPUI associated with the dismiss event.
+     */
+    data class Dismiss<T : AEPUITemplate, S : AEPUIState>(val _aepui: AEPUI<T, S>) : UIEvent<T, S>
+
+    /**
+     * Event that represents the expansion of a UI element.
+     *
+     * @param T The type of UI template associated with the expand event.
+     * @param S The type of UI state associated with the expand event.
+     * @property _aepui The AEPUI associated with the expand event.
+     */
+    data class Expand<T : AEPUITemplate, S : AEPUIState>(val _aepui: AEPUI<T, S>) : UIEvent<T, S>
```

Comment:
This needs to be changed to `Display`, `Dismiss` and `Interact` (with an `Action` to specify  the type of interaction)